src/pykeen/nn/utils.py

```diff
@@ -324,3 +326,42 @@ def get_descriptions(self, ids: Sequence[str]) -> Sequence[str]:
             the description for each Wikidata entity
         """
         return self._get(ids=ids, component="description")
+
+
+class ShapeError(ValueError):
+    """An error for a mismatch in shapes."""
+
+    def __init__(self, shape: OneOrSequence[int], reference: OneOrSequence[int]) -> None:
```

Comment:
Not that it has a huge effect, but these have already been pre-upgraded to sequences in the `verify()` so you could make it more strict to `Sequence[int]` in both

```suggestion
    def __init__(self, shape: Sequence[int], reference: Sequence[int]) -> None:
```